RoutesApp_ios/utilities/helpers/Algorithm.swift

```diff
@@ -0,0 +1,164 @@
+                originRoutes.append(newLine)
+            }
+        }
+        return originRoutes
+    }
+
+    private func getDestinationRoutes(destination: CLLocationCoordinate2D, lines: [Line], minDistanceBtwStops: Double) -> [Line] {
+        var destinationRoutes = [Line]()
+        for line in lines {
+            let nearestStopToDestination = line.stops.min(by: {
+                $0.toCLLocationCoordinate2D().distance(to: destination) < $1.toCLLocationCoordinate2D().distance(to: destination)
+            })
+
+            if let nearestStopToDestination = nearestStopToDestination,
+               nearestStopToDestination.toCLLocationCoordinate2D().distance(to: destination) <= minDistanceBtwStops {
+                // MARK: Slice Stops
+                let indexStopDestination = getIndexWhere(coordinate: nearestStopToDestination, coordinateList: line.stops)
+                let newStops = Array(line.stops[0...indexStopDestination])
+
+                // MARK: Slice Route points
+                let indexDestinationPoint = getIndexWhere(coordinate: nearestStopToDestination, coordinateList: line.routePoints)
+                let newRoutePoints = Array(line.routePoints[...indexDestinationPoint])
+
+                let newLine = Line(name: line.name, categoryRef: line.categoryRef, routePoints: newRoutePoints, start: line.start, stops: newStops)
+                destinationRoutes.append(newLine)
+            }
+        }
+        return destinationRoutes
+    }
+
+    private func getCombinedRoutes(origin: CLLocationCoordinate2D, destination: CLLocationCoordinate2D,
+                                   candidates: LinesCandidate, minDistanceBtwPoints: Double, minDistanceBtwStops: Double) -> [AvailableTransport] {
+        var availableTransports: [AvailableTransport] = []
+        for routeFromOrigin in candidates.originList {
+            for routeFromDestination in candidates.destinationList {
+                if routeFromOrigin == routeFromDestination {
+                    continue
+                }
+                for stop in routeFromDestination.stops {
+                    let nearestStopToOrigin = routeFromOrigin.stops.min(by: {
+                        $0.toCLLocationCoordinate2D().distance(to: stop.toCLLocationCoordinate2D()) <=
+                            $1.toCLLocationCoordinate2D().distance(to: stop.toCLLocationCoordinate2D())
+                    })
+                    guard let nearestStopToOrigin = nearestStopToOrigin else { continue }
+                    let indexOfNearestStopToOrigin = getIndexWhere(coordinate: nearestStopToOrigin, coordinateList: routeFromOrigin.stops)
+                    if nearestStopToOrigin.toCLLocationCoordinate2D().distance(to: stop.toCLLocationCoordinate2D()) <= minDistanceBtwStops {
+                        // MARK: Origin line splice
+                        let indexOfOriginRoute = getIndexWhere(coordinate: nearestStopToOrigin, coordinateList: routeFromOrigin.routePoints)
```

Comment:
The other methods are great but this one is a little bit bigger.
Try to reduce cognitive complexity inside this if, call a couple of methods to obtain the values that you´re calculating here.

Reply:
as a hint, every MARK you´re using can be separated as a method

Reply:
Added more methods to make it easier to understand